parts/windows/windowscontainerdfunc.ps1

```diff
@@ -37,9 +37,13 @@ function RegisterContainerDService {
   if ($null -eq $svc) {
     Set-ExitCode -ExitCode $global:WINDOWS_CSE_ERROR_CONTAINERD_NOT_INSTALLED -ErrorMessage "containerd.exe did not get installed as a service correctly."
   }
-  Start-Service containerd
   if ($svc.Status -ne "Running") {
-    Set-ExitCode -ExitCode $global:WINDOWS_CSE_ERROR_CONTAINERD_NOT_RUNNING -ErrorMessage "containerd service is not running"
+    try {
+      Start-Service containerd
+    }
+    catch {

```

Comment:
Does it mean start-service will throw an exception? Could we retry 3 times to start-service and check the status, then return the error if all retires fail?

Reply:
start-service throws an exception in some cases. I am not sure we are over-protecting the service. BTW, docker is using the same way to start up, but yeah we may also have issues in starting docker.

Reply:
I did not see such error with docker, do you?
Please add an error if any exception will be thrown in starting containerd. And `Set-ExitCode -ExitCode $global:WINDOWS_CSE_ERROR_CONTAINERD_NOT_RUNNING -ErrorMessage $_` after checking that containerd is not running

Reply:
I did not see such error with docker, do you?
Please add an error if any exception will be thrown in starting containerd. And `Set-ExitCode -ExitCode $global:WINDOWS_CSE_ERROR_CONTAINERD_NOT_RUNNING -ErrorMessage $_` after checking that containerd is not running

Reply:
done, please take a look.